content/post/postgres-wal-nfs/index.md

```diff
@@ -0,0 +1,506 @@
+To get cluster ID we can run the following command:
+
+```bash
+$ kubectl get ns kube-system -o jsonpath='{.metadata.uid}'
+60b010fb-9ad6-4ac6-89f4-7321e697f469
+```
+
+### Get License
+
+Go to [Appscode License Server](https://license-issuer.appscode.com/) to get the license.txt file. For this tutorial we will use KubeDB Enterprise Edition.
+
+![License Server](AppscodeLicense.png)
+
+As we have license, now we will follow [this](https://github.com/kubedb/archiver-demo) to install the necessary  tools for performing PITR.
+
+### Install Sidekick
+
+```bash
+helm pull oci://ghcr.io/appscode-charts/sidekick --version v2023.12.11
+```
+```bash
+helm upgrade -i sidekick sidekick-v2023.12.11.tgz \
+  -n kubedb --create-namespace \
+  --wait --burst-limit=10000 --debug
+```
+![Sidekick image](sidekick.png)
+
+### Install KubeStash
+```bash
+helm pull oci://ghcr.io/appscode-charts/kubestash --version v2023.12.28
+```
+> Please provide the license that you have downloaded earlier here `--set-file global.license=`
+
+
+```bash
+helm upgrade -i kubestash kubestash-v2023.12.28.tgz \
+  --namespace kubedb \
+  --set kubedb-kubestash-catalog.enabled=true \
+
+  --set-file global.license=$HOME/Downloads/kubedb-license-6352bd41-0d2d-4824-a70c-cd038fe8292d.txt \
+  --wait --burst-limit=10000 --debug
+```
+
+![KubeStash](kubestash.png)
+
+### Install Kubedb
+
+Uninstall kubedb if already kubedb relase exists.
+```bash
+helm uninstall kubedb -n kubedb
+```
+
+```bash
+helm pull oci://ghcr.io/appscode-charts/kubedb --version v2023.12.28
+```
+
+Apply the crds.
+```bash
+kubectl apply -f https://github.com/kubedb/installer/raw/v2023.12.28/crds/kubedb-catalog-crds.yaml
+
+```
+
+> Please provide the license that you have downloaded earlier here `--set-file global.license=`
+
+
+```bash
+helm upgrade -i kubedb kubedb-v2023.12.28.tgz \
+  --namespace kubedb \
+  --set kubedb-kubestash-catalog.enabled=true \
+  --set-file global.license=$HOME/Downloads/kubedb-license-6352bd41-0d2d-4824-a70c-cd038fe8292d.txt \
+  --wait --burst-limit=10000 --debug
+```
+
+![kubedb image](kubedb.png)
+
+### Install VolumeSnapshotter
+If your cluster already have a volume-snapshotter available, You don't need to follow this step.
+Otherwise, You need to install an external-snapshotter to take volume snapshot.
+
+clone https://github.com/kubernetes-csi/external-snapshotter/tree/release-5.0
+
+or 
+
+wget https://github.com/kubernetes-csi/external-snapshotter/archive/refs/tags/v5.0.1.zip
+
+```bash
+unzip v5.0.1.zip
+cd external-snapshotter-5.0.1
+
+kubectl kustomize client/config/crd | kubectl create -f -
+kubectl -n kube-system kustomize deploy/kubernetes/snapshot-controller | kubectl create -f -
+kubectl kustomize deploy/kubernetes/csi-snapshotter | kubectl create -f -
+```
+
+
+### Install Longhorn
+You will need to install longhorn for provisioning volume snapshot. If you already have a volume snapshot provisioner available in your cluster, ignore this step.
```

Comment:
### Install Longhorn
Longhorn is a distributed block storage system for Kubernetes that manages persistent storage. 

### Install External Snapshotter
CSI (Container Storage Interface) external snapshotter is a component used in Kubernetes for managing volume snapshots. It allows external storage systems (such as Longhorn) to integrate with Kubernetes Volume Snapshot feature.